src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -1,87 +1,217 @@
         public Uri Uri { get; set; }
 
         /// <summary>
-        /// Optional Http Body
+        ///     Optional Http Body
         /// </summary>
-        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 3, HelpMessage = "Request Body. Required when Method is Post or Patch")]
+        [Parameter(ParameterSetName = Constants.UserParameterSet,
+            Position = 3,
+            HelpMessage = "Request Body. Required when Method is Post or Patch",
+            ValueFromPipeline = true)]
         public object Body { get; set; }
 
         /// <summary>
-        /// Wait for .NET debugger to attach
+        ///     Optional Custom Headers
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 4,
+            HelpMessage = "Optional Custom Headers")]
+        public IDictionary<string, string> Headers { get; set; }
+
+        /// <summary>
+        ///     Output file where the response body will be saved
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 5,
+            HelpMessage = "Output file where the response body will be saved")]
+        public string OutFile { get; set; }
+
+        /// <summary>
+        ///     Infer Download FileName from ContentDisposition Header
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 6,
+            HelpMessage = "Infer Download FileName")]
+        public SwitchParameter InferOutFileName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the InFile property.
+        /// </summary>
+        [Parameter]
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 7,
+            HelpMessage = "Infile to Send in the Request")]
+        public virtual string InFile { get; set; }
+
+        /// <summary>
+        ///     Indicates that the cmdlet returns the results, in addition to writing them to a file.
+        ///     only valid when the OutFile parameter is also used.
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 8,
+            HelpMessage =
+                "Indicates that the cmdlet returns the results, in addition to writing them to a file. Only valid when the OutFile parameter is also used. ")]
+        public SwitchParameter PassThru { get; set; }
+
+        /// <summary>
+        ///     OAuth or Bearer Token to use instead of already acquired token
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = Constants.UserParameterSet,
+            Position = 9,
+            HelpMessage = "OAuth or Bearer Token to use instead of already acquired token")]
+        public string Token { get; set; }
```

Comment:
Update the setter to set 

`Authentication = GraphRequestAuthenticationType.UserProvidedToken`

It currently isn't set anywhere and later logic uses this.

Reply:
In the constructor its set to Default.